File: app/src/main/java/com/ramez/shopp/Classes/GlobalData.java
package com.ramez.shopp.Classes;

import android.content.Context;
import android.widget.ImageView;
import android.widget.Toast;

import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeErrorDialog;
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeInfoDialog;
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeProgressDialog;
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeSuccessDialog;
import com.bumptech.glide.Glide;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;
import com.bumptech.glide.request.RequestOptions;
import com.bumptech.glide.request.target.Target;
import com.ramez.shopp.Models.CountryModel;
import com.ramez.shopp.R;
import com.squareup.picasso.Callback;
import com.squareup.picasso.NetworkPolicy;
import com.squareup.picasso.Picasso;

import java.util.HashMap;
import java.util.Map;


public class GlobalData {

    public static final String BetaBaseURL = "https://risteh.com/";
    public static final String grocery = "/GroceryStoreApi/";
    public static final String BaseURL = BetaBaseURL;
    //public static final String ApiURL = BaseURL + "api/";
    public static final String Api = "api/";
    public static final String COUNTRY = "BH";
    public static final Boolean refresh_cart = false;
    public static Boolean refresh_points = false;


    public static int Position = 0;

    public static boolean EDIT_PROFILE = false;
    public static boolean REFRESH_ADV = false;
    public static boolean REFRESH_CART = false;
    public static int CHAT_ID_OPEN = 0;
    public static AwesomeErrorDialog errorDialog;
    public static AwesomeInfoDialog infoDialog;
    public static AwesomeSuccessDialog successDialog;
    static AwesomeProgressDialog progressDialog;

    //============================================================================

    public static void PicassoImg(String image, int placeholder, ImageView imageView) {
        String photoUrl = "";

        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        String finalPhotoUrl = photoUrl;

        Picasso.get().load(finalPhotoUrl).placeholder(placeholder).error(R.drawable.holder_image).into(imageView);

    }


    public static void PicassoOffLineImg(String image, int placeholder, ImageView imageView) {
        String photoUrl = "";

        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        String finalPhotoUrl = photoUrl;
        Picasso.get().load(photoUrl).placeholder(placeholder).error(R.drawable.holder_image).networkPolicy(NetworkPolicy.OFFLINE).into(imageView, new Callback() {
            @Override
            public void onSuccess() {

            }

            @Override
            public void onError(Exception e) {
                Picasso.get().load(finalPhotoUrl).placeholder(placeholder).error(R.drawable.holder_image).into(imageView);
            }
        });

    }

    public static void GlideImg(Context c, String image, int placeholder, ImageView imageView) {
        String photoUrl = "";
        RequestOptions requestOptions = new RequestOptions();
        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        try {
            Glide.with(c).asBitmap().load(photoUrl)
                    .apply(requestOptions)
                    .thumbnail(0.5f)
                    .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                    .placeholder(placeholder).into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }


    public static void GlideImgGif(Context c, String image, int placeholder, ImageView imageView) {
        String photoUrl = "";

        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        try {
            Glide.with(c).load(photoUrl).override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL).placeholder(placeholder).into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }


    }


    public static void GlideImgWithTransform(Context c, String image, int placeholder, ImageView imageView) {
        String photoUrl = "";

        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        try {
            Glide.with(c).load(photoUrl)
                    .transform(new RoundedCorners(25))
                    .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                    .placeholder(placeholder).into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }


    }

    public static void GlideImgWeb(Context c, String image, int placeholder, ImageView imageView) {
        String photoUrl = "";

        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        try {

            Glide.with(c).load(photoUrl)
//                    .apply(requestOptions)
//                    .apply(RequestOptions.bitmapTransform(new RoundedCorners(14)))
                    .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL).placeholder(placeholder).into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }


    public static void GlideImgGifSize(Context c, String image, int placeholder, ImageView imageView) {
        String photoUrl = "";

        if (image != null && !image.isEmpty()) {
            photoUrl = image;
        } else {
            photoUrl = BetaBaseURL;
        }

        try {
            Glide.with(c).load(photoUrl).placeholder(placeholder).apply(new RequestOptions().transform(new RoundedCorners((int) c.getResources().getDimension(R.dimen._8sdp))).error(R.drawable.holder_image).skipMemoryCache(true).diskCacheStrategy(DiskCacheStrategy.NONE))
                    .into(imageView);
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public static void progressDialog(Context c, int title, int msg) {
        progressDialog = new AwesomeProgressDialog(c);
        progressDialog.setTitle(title).setMessage(msg).setColoredCircle(R.color.colorPrimaryDark).setDialogIconAndColor(R.drawable.ic_dialog_info, R.color.white).setCancelable(false);
        progressDialog.show();

    }

    public static void hideProgressDialog() {
        if (progressDialog != null) progressDialog.hide();

    }

    public static void errorDialog(Context c, int title, String msg) {
        errorDialog = new AwesomeErrorDialog(c);
        errorDialog.setTitle(title);
        errorDialog.setMessage(msg);
        errorDialog.setColoredCircle(R.color.dialogErrorBackgroundColor).setDialogIconAndColor(R.drawable.ic_dialog_error, R.color.white).setCancelable(true).setButtonBackgroundColor(R.color.dialogErrorBackgroundColor);
        errorDialog.show();

    }


    public static void errorDialogWithButton(Context c, String title, String msg) {
        errorDialog = new AwesomeErrorDialog(c);
        errorDialog.setTitle(title);
        errorDialog.setMessage(msg);
        errorDialog.setColoredCircle(R.color.dialogErrorBackgroundColor).
                setDialogIconAndColor(R.drawable.ic_dialog_error, R.color.white).setCancelable(true).
                setButtonBackgroundColor(R.color.dialogErrorBackgroundColor).setButtonText(c.getString(R.string.ok)).setErrorButtonClick(() -> errorDialog.hide())

        ;
        errorDialog.show();

    }

    public static void infoDialog(Context c, String title, String msg) {
        infoDialog = new AwesomeInfoDialog(c);
        infoDialog.setMessage(msg);
        infoDialog.setTitle(title);
        infoDialog.setColoredCircle(R.color.dialogInfoBackgroundColor).setDialogIconAndColor(R.drawable.ic_dialog_info, R.color.white).setCancelable(true);
        infoDialog.show();

    }


    public static void successDialog(Context c, String title, String msg) {
        successDialog = new AwesomeSuccessDialog(c);
        successDialog.setTitle(title).setMessage(msg).setColoredCircle(R.color.dialogSuccessBackgroundColor).setDialogIconAndColor(R.drawable.ic_check, R.color.white).setCancelable(true);
        successDialog.show();

    }


    public static void Toast(Context context, String msg) {

        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show();
    }

    public static void Toast(Context context, int resId) {

        Toast.makeText(context, context.getString(resId), Toast.LENGTH_SHORT).show();
    }

    public static String getIntro(String countryCode) {
        String introStr = "3";
        HashMap<String, String> hashMap = new HashMap<String, String>();

        // country Id :4 ,country name Oman
        // country Id :17 ,country name Bahrain
        // country Id :117 ,country name Kuwait
        // country Id :178 ,country name Qatar
        // country Id :191 ,country name Saudi_Arabia
        // country Id :229 ,country name United_Arab_Emirates_ar

        hashMap.put("968", "9");
        hashMap.put("973", "3");
        hashMap.put("965", "6");
        hashMap.put("974", "5");
        hashMap.put("966", "5");
        hashMap.put("971", "5");

        if(countryCode.equals("966")||countryCode.equals("971")){
            introStr=hashMap.get(countryCode)+"xxxxxxxxx";

        }
        else {
            introStr=hashMap.get(countryCode)+ "xxxxxxxx";

        }

        return introStr;
    }


}

File: app/src/main/java/com/ramez/shopp/Classes/GlobalData.kt
package com.ramez.shopp.Classes


import android.content.Context
import android.widget.ImageView
import android.widget.Toast
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeErrorDialog
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeInfoDialog
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeProgressDialog
import com.awesomedialog.blennersilva.awesomedialoglibrary.AwesomeSuccessDialog
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.load.resource.bitmap.RoundedCorners
import com.bumptech.glide.request.RequestOptions
import com.bumptech.glide.request.target.Target
import com.ramez.shopp.R
import com.squareup.picasso.Callback
import com.squareup.picasso.NetworkPolicy
import com.squareup.picasso.Picasso
import java.lang.Exception
import java.util.HashMap


object GlobalData {
    const val BetaBaseURL = "https://risteh.com/"
    const val grocery = "/GroceryStoreApi/"
    const val BaseURL = BetaBaseURL

    //public static final String ApiURL = BaseURL + "api/";
    const val Api = "api/"
    const val COUNTRY = "BH"
    var refresh_cart = false
    var refresh_points = false
    var Position = 0
    var EDIT_PROFILE = false
    var REFRESH_ADV = false
    var REFRESH_CART = false
    var CHAT_ID_OPEN = 0
    var errorDialog: AwesomeErrorDialog? = null
    var infoDialog: AwesomeInfoDialog? = null
    var successDialog: AwesomeSuccessDialog? = null
    var progressDialog: AwesomeProgressDialog? = null

    //============================================================================
    fun PicassoImg(image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        val finalPhotoUrl = photoUrl
        Picasso.get().load(finalPhotoUrl).placeholder(placeholder).error(R.drawable.holder_image)
            .into(imageView)
    }

    fun PicassoOffLineImg(image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        val finalPhotoUrl = photoUrl
        Picasso.get().load(photoUrl).placeholder(placeholder).error(R.drawable.holder_image)
            .networkPolicy(NetworkPolicy.OFFLINE).into(imageView, object : Callback {
                override fun onSuccess() {}
                override fun onError(e: Exception) {
                    Picasso.get().load(finalPhotoUrl).placeholder(placeholder).error(R.drawable.holder_image)
                        .into(imageView)
                }
            })
    }

    fun GlideImg(c: Context?, image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        val requestOptions = RequestOptions()
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        try {
            Glide.with(c!!).asBitmap().load(photoUrl)
                .apply(requestOptions)
                .thumbnail(0.5f)
                .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                .placeholder(placeholder).into(imageView!!)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun GlideImgGif(c: Context?, image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        try {
            Glide.with(c!!).load(photoUrl).override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                .placeholder(placeholder).into(
                    imageView!!
                )
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun GlideImgWithTransform(c: Context?, image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        try {
            Glide.with(c!!).load(photoUrl)
                .transform(RoundedCorners(25))
                .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL)
                .placeholder(placeholder).into(imageView!!)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun GlideImgWeb(c: Context?, image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        try {
            Glide.with(c!!).load(photoUrl) //                    .apply(requestOptions)
                //                    .apply(RequestOptions.bitmapTransform(new RoundedCorners(14)))
                .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL).placeholder(placeholder).into(
                    imageView!!
                )
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun GlideImgGifSize(c: Context, image: String?, placeholder: Int, imageView: ImageView?) {
        var photoUrl = ""
        photoUrl = if (image != null && !image.isEmpty()) {
            image
        } else {
            BetaBaseURL
        }
        try {
            Glide.with(c).load(photoUrl).placeholder(placeholder).apply(
                RequestOptions().transform(
                    RoundedCorners(
                        c.resources.getDimension(R.dimen._8sdp).toInt()
                    )
                ).error(R.drawable.holder_image).skipMemoryCache(true)
                    .diskCacheStrategy(DiskCacheStrategy.NONE)
            )
                .into(imageView!!)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun progressDialog(c: Context?, title: Int, msg: Int) {
        progressDialog = AwesomeProgressDialog(c)
        progressDialog!!.setTitle(title).setMessage(msg).setColoredCircle(R.color.colorPrimaryDark)
            .setDialogIconAndColor(R.drawable.ic_dialog_info, R.color.white).setCancelable(false)
        progressDialog!!.show()
    }

    fun hideProgressDialog() {
        if (progressDialog != null) progressDialog!!.hide()
    }

    fun errorDialog(c: Context?, title: Int, msg: String?) {
        errorDialog = AwesomeErrorDialog(c)
        errorDialog!!.setTitle(title)
        errorDialog!!.setMessage(msg)
        errorDialog!!.setColoredCircle(R.color.dialogErrorBackgroundColor)
            .setDialogIconAndColor(R.drawable.ic_dialog_error, R.color.white).setCancelable(true)
            .setButtonBackgroundColor(R.color.dialogErrorBackgroundColor)
        errorDialog!!.show()
    }

    fun errorDialogWithButton(c: Context, title: String?, msg: String?) {
        errorDialog = AwesomeErrorDialog(c)
        errorDialog!!.setTitle(title)
        errorDialog!!.setMessage(msg)
        errorDialog!!.setColoredCircle(R.color.dialogErrorBackgroundColor)
            .setDialogIconAndColor(R.drawable.ic_dialog_error, R.color.white).setCancelable(true)
            .setButtonBackgroundColor(R.color.dialogErrorBackgroundColor)
            .setButtonText(c.getString(R.string.ok)).setErrorButtonClick { errorDialog!!.hide() }
        errorDialog!!.show()
    }

    fun infoDialog(c: Context?, title: String?, msg: String?) {
        infoDialog = AwesomeInfoDialog(c)
        infoDialog!!.setMessage(msg)
        infoDialog!!.setTitle(title)
        infoDialog!!.setColoredCircle(R.color.dialogInfoBackgroundColor)
            .setDialogIconAndColor(R.drawable.ic_dialog_info, R.color.white).setCancelable(true)
        infoDialog!!.show()
    }

    fun successDialog(c: Context?, title: String?, msg: String?) {
        successDialog = AwesomeSuccessDialog(c)
        successDialog!!.setTitle(title).setMessage(msg).setColoredCircle(R.color.dialogSuccessBackgroundColor)
            .setDialogIconAndColor(R.drawable.ic_check, R.color.white).setCancelable(true)
        successDialog!!.show()
    }

    fun Toast(context: Context?, msg: String?) {
        Toast.makeText(context, msg, Toast.LENGTH_SHORT).show()
    }

    fun Toast(context: Context, resId: Int) {
        Toast.makeText(context, context.getString(resId), Toast.LENGTH_SHORT).show()
    }

    fun getIntro(countryCode: String): String {
        var introStr = "3"
        val hashMap = HashMap<String, String>()

        // country Id :4 ,country name Oman
        // country Id :17 ,country name Bahrain
        // country Id :117 ,country name Kuwait
        // country Id :178 ,country name Qatar
        // country Id :191 ,country name Saudi_Arabia
        // country Id :229 ,country name United_Arab_Emirates_ar
        hashMap["968"] = "9"
        hashMap["973"] = "3"
        hashMap["965"] = "6"
        hashMap["974"] = "5"
        hashMap["966"] = "5"
        hashMap["971"] = "5"
        introStr = if (countryCode == "966" || countryCode == "971") {
            hashMap[countryCode].toString() + "xxxxxxxxx"
        } else {
            hashMap[countryCode].toString() + "xxxxxxxx"
        }
        return introStr
    }
}
